### ISSUE TITLE
Multiline not working, unable to reproduce documented example 

### ISSUE THREAD

**Issue by amitoj** [fluent/fluent-bit#4595]
Trying to replicate the example from https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/multiline-parsing
and unable to get the multiline parsing working

I'm using v1.9.0, binary compiled from source (MacBook Pro M1)
```
% fluent-bit --version
Fluent Bit v1.9.0
```

My output is

```
% fluent-bit -c fluent-bit.conf
Fluent Bit v1.9.0
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/01/10 21:15:04] [ info] [engine] started (pid=27833)
[2022/01/10 21:15:04] [ info] [storage] version=1.1.5, initializing...
[2022/01/10 21:15:04] [ info] [storage] in-memory
[2022/01/10 21:15:04] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/01/10 21:15:04] [ info] [cmetrics] version=0.2.2
[2022/01/10 21:15:04] [ info] [input:tail:tail.0] multiline core started
[2022/01/10 21:15:04] [ info] [sp] stream processor started
[0] tail.0: [1641849304.145157000, {"log"=>"single line...
"}]
[1] tail.0: [1641849304.145157000, {"log"=>"Dec 14 06:41:08 Exception in thread "main" java.lang.RuntimeException: Something has gone wrong, aborting!
"}]
[2] tail.0: [1641849304.145157000, {"log"=>"    at com.myproject.module.MyProject.badMethod(MyProject.java:22)
"}]
[3] tail.0: [1641849304.145157000, {"log"=>"    at com.myproject.module.MyProject.oneMoreMethod(MyProject.java:18)
"}]
[4] tail.0: [1641849304.145157000, {"log"=>"    at com.myproject.module.MyProject.anotherMethod(MyProject.java:14)
"}]
[5] tail.0: [1641849304.145157000, {"log"=>"    at com.myproject.module.MyProject.someMethod(MyProject.java:10)
"}]
[6] tail.0: [1641849304.145157000, {"log"=>"    at com.myproject.module.MyProject.main(MyProject.java:6)
"}]
[7] tail.0: [1641849304.145157000, {"log"=>"another line...
"}]
```

whereas I'm expecting,

```
$ fluent-bit -c fluent-bit.conf 

[0] tail.0: [0.000000000, {"log"=>"single line...
"}]
[1] tail.0: [1626634867.472226330, {"log"=>"Dec 14 06:41:08 Exception in thread "main" java.lang.RuntimeException: Something has gone wrong, aborting!
    at com.myproject.module.MyProject.badMethod(MyProject.java:22)
    at com.myproject.module.MyProject.oneMoreMethod(MyProject.java:18)
    at com.myproject.module.MyProject.anotherMethod(MyProject.java:14)
    at com.myproject.module.MyProject.someMethod(MyProject.java:10)
    at com.myproject.module.MyProject.main(MyProject.java:6)
"}]
[2] tail.0: [1626634867.472226330, {"log"=>"another line...
"}]
```

**Comment by patrick-stephens** [comment]
Yeah I've repeated this with the containers as well, it does seem to be on the `master` branch. 
I only added absolute paths for the various files (parser and input log) are set to `/fluent-bit/etc/` to simplify mounting into the container.
```
$ docker run --rm -v $PWD:/fluent-bit/etc/ ghcr.io/fluent/fluent-bit
Fluent Bit v1.8.11
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/01/11 20:32:52] [ info] [engine] started (pid=1)
[2022/01/11 20:32:52] [ info] [storage] version=1.1.5, initializing...
[2022/01/11 20:32:52] [ info] [storage] in-memory
[2022/01/11 20:32:52] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/01/11 20:32:52] [ info] [cmetrics] version=0.2.2
[2022/01/11 20:32:52] [ info] [input:tail:tail.0] multiline core started
[2022/01/11 20:32:52] [ info] [sp] stream processor started
[2022/01/11 20:32:52] [ info] [input:tail:tail.0] inotify_fs_add(): inode=100541195 watch_fd=1 name=/fluent-bit/etc/test.log
[0] tail.0: [1641933172.917577689, {"log"=>"single line...
"}]
[1] tail.0: [1641933172.917595777, {"log"=>"Dec 14 06:41:08 Exception in thread "main" java.lang.RuntimeException: Something has gone wrong, aborting!
    at com.myproject.module.MyProject.badMethod(MyProject.java:22)
    at com.myproject.module.MyProject.oneMoreMethod(MyProject.java:18)
    at com.myproject.module.MyProject.anotherMethod(MyProject.java:14)
    at com.myproject.module.MyProject.someMethod(MyProject.java:10)
    at com.myproject.module.MyProject.main(MyProject.java:6)
"}]
[2] tail.0: [1641933172.917595777, {"log"=>"another line...
"}]
^C[2022/01/11 20:32:55] [engine] caught signal (SIGINT)
[2022/01/11 20:32:55] [ info] [input] pausing tail.0
[2022/01/11 20:32:55] [ warn] [engine] service will shutdown in max 5 seconds
[2022/01/11 20:32:56] [ info] [engine] service has stopped (0 pending tasks)
[2022/01/11 20:32:56] [ info] [input:tail:tail.0] inotify_fs_remove(): inode=100541195 watch_fd=1

$ docker run --rm -v $PWD:/fluent-bit/etc/ ghcr.io/fluent/fluent-bit:x86_64-master
Fluent Bit v1.9.0
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/01/11 20:33:00] [ info] [engine] started (pid=1)
[2022/01/11 20:33:00] [ info] [storage] version=1.1.5, initializing...
[2022/01/11 20:33:00] [ info] [storage] in-memory
[2022/01/11 20:33:00] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/01/11 20:33:00] [ info] [cmetrics] version=0.2.2
[2022/01/11 20:33:00] [ info] [input:tail:tail.0] multiline core started
[2022/01/11 20:33:00] [ info] [sp] stream processor started
[2022/01/11 20:33:00] [ info] [input:tail:tail.0] inotify_fs_add(): inode=100541195 watch_fd=1 name=/fluent-bit/etc/test.log
[0] tail.0: [1641933180.524041014, {"log"=>"single line...
"}]
[1] tail.0: [1641933180.524041014, {"log"=>"Dec 14 06:41:08 Exception in thread "main" java.lang.RuntimeException: Something has gone wrong, aborting!
"}]
[2] tail.0: [1641933180.524041014, {"log"=>"    at com.myproject.module.MyProject.badMethod(MyProject.java:22)
"}]
[3] tail.0: [1641933180.524041014, {"log"=>"    at com.myproject.module.MyProject.oneMoreMethod(MyProject.java:18)
"}]
[4] tail.0: [1641933180.524041014, {"log"=>"    at com.myproject.module.MyProject.anotherMethod(MyProject.java:14)
"}]
[5] tail.0: [1641933180.524041014, {"log"=>"    at com.myproject.module.MyProject.someMethod(MyProject.java:10)
"}]
[6] tail.0: [1641933180.524041014, {"log"=>"    at com.myproject.module.MyProject.main(MyProject.java:6)
"}]
[7] tail.0: [1641933180.524041014, {"log"=>"another line...
"}]
^C[2022/01/11 20:33:03] [engine] caught signal (SIGINT)
[2022/01/11 20:33:03] [ info] [input] pausing tail.0
[2022/01/11 20:33:03] [ warn] [engine] service will shutdown in max 5 seconds
[2022/01/11 20:33:04] [ info] [engine] service has stopped (0 pending tasks)
[2022/01/11 20:33:04] [ info] [input:tail:tail.0] inotify_fs_remove(): inode=100541195 watch_fd=1
```

**Comment by ismailazerty** [comment]
sorry but where did you find the v1.9.0 , I only find v1.8.11 ? thx

**Comment by patrick-stephens** [comment]
Compiled from source.

**Comment by patrick-stephens** [comment]
This seems to be resolved now for 1.9.2, repeating the same test as before:
```
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/04/22 09:47:08] [ info] [fluent bit] version=1.9.2, commit=27a63c11d3, pid=1
[2022/04/22 09:47:08] [ info] [storage] version=1.1.6, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/04/22 09:47:08] [ info] [cmetrics] version=0.3.0
[2022/04/22 09:47:08] [ info] [input:tail:tail.0] multiline core started
[2022/04/22 09:47:08] [ info] [sp] stream processor started
[2022/04/22 09:47:08] [ info] [output:stdout:stdout.0] worker #0 started
[2022/04/22 09:47:08] [ info] [input:tail:tail.0] inotify_fs_add(): inode=100270112 watch_fd=1 name=/fluent-bit/etc/test.log
[0] tail.0: [1650620828.029681320, {"log"=>"single line...
"}]
[1] tail.0: [1650620828.029726157, {"log"=>"Dec 14 06:41:08 Exception in thread "main" java.lang.RuntimeException: Something has gone wrong, aborting!
    at com.myproject.module.MyProject.badMethod(MyProject.java:22)
    at com.myproject.module.MyProject.oneMoreMethod(MyProject.java:18)
    at com.myproject.module.MyProject.anotherMethod(MyProject.java:14)
    at com.myproject.module.MyProject.someMethod(MyProject.java:10)
    at com.myproject.module.MyProject.main(MyProject.java:6)
"}]
[2] tail.0: [1650620828.029726157, {"log"=>"another line...
"}]

```
